source/common/http/conn_manager_impl.cc

```diff
@@ -987,6 +965,11 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
   ASSERT(!cached_route_);
   refreshCachedRoute();
 
+  auto req_resp_stats = clusterRequestResponseSizeStats();
+  if (req_resp_stats.has_value()) {
+    req_resp_stats->get().upstream_rq_headers_size_.recordValue(request_headers_->byteSize());
+  }
+
```

Comment:
Note to self & others (will add a inline comment most likely): an important change here is that we now compute request headers size _after_ the route has been selected (because that's what makes cluster info available), so we are most likely adding some internally added headers to the calculation. We might want to note this in some related docs as well.